typed-racket-lib/typed-racket/static-contracts/structures.rkt

```diff
@@ -9,6 +9,10 @@
 
 (provide prop:combinator-name gen:sc)
 
+(provide static-contract-may-contain-free-ids?)
+
+(define static-contract-may-contain-free-ids? (make-parameter #f))
```

Comment:
Could this be a free id set (of the ids that can appear "free") instead of a boolean?

Reply:
What would be ideal is if static-contracts knew what their set of free ids were -- then we wouldn't need a parameter. Without that info... I think this is the best we can do.

I added a comment describing why we're using the parameter and the note about if static-contracts knew their free ids this would be unnecessary.